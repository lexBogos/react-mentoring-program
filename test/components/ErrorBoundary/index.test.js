import React from 'react';
import ErrorBoundary from '../../../app/Components/ErrorBoundary/index';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme,{ mount } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';

Enzyme.configure({ adapter: new Adapter() });

const Something = () => <h1>Smth went wrong!</h1>;

describe('ErrorBoundary component', () => {
    it('should display an ErrorMessage if wrapped component throws', () => {
        const wrapper = mount(
          <ErrorBoundary>
            <Something />
          </ErrorBoundary>
        );
    
        const error = new Error('test');
    
        expect(wrapper.find(Something).simulateError(error).debug()).toMatchSnapshot();
       
      })
})
