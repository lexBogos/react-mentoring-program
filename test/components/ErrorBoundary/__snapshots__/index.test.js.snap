// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component should display an ErrorMessage if wrapped component throws 1`] = `
"<Something>
  <h1>
    Smth went wrong!
  </h1>
</Something>"
`;
